File: notification_hub_backend_service/src/xamarin/PushDemo/PushDemo/Models/DeviceInstallation.cs
using System.Collections.Generic;
using Newtonsoft.Json;

namespace PushDemo.Models
{
    public class DeviceInstallation
    {
        [JsonProperty("installationId")]
        public string InstallationId { get; set; }

        [JsonProperty("platform")]
        public string Platform { get; set; }

        [JsonProperty("pushChannel")]
        public string PushChannel { get; set; }

        [JsonProperty("tags")]
        public List<string> Tags { get; set; } = new List<string>();
    }
}

File: notification_hub_backend_service/src/flutter/push_demo/lib/models/device_installation.dart
class DeviceInstallation {
  final String deviceId;
  final String platform;
  final String token;
  final List<String> tags;

  DeviceInstallation(this.deviceId, this.platform, this.token, this.tags);

  DeviceInstallation.fromJson(Map<String, dynamic> json)
      : deviceId = json['installationId'],
        platform = json['platform'],
        token = json['pushChannel'],
        tags = json['tags'];

  Map<String, dynamic> toJson() =>
    {
      'installationId': deviceId,
      'platform': platform,
      'pushChannel': token,
      'tags': tags,
    };
}